pkg/services/folder/folderimpl/folder.go

```diff
@@ -188,11 +190,55 @@ func (s *Service) GetChildren(ctx context.Context, cmd *folder.GetChildrenQuery)
 		childrenUIDs = append(childrenUIDs, f.UID)
 	}
 
+	availableNonRootFolders := make([]*folder.Folder, 0)
+	if cmd.UID == "" {
+		availableNonRootFolders, err = s.getAvailableNonRootFolders(ctx, cmd.OrgID, cmd.SignedInUser)
+		if err != nil {
+			return nil, folder.ErrInternal.Errorf("failed to fetch subfolders from dashboard store: %w", err)
+		}
+
+		for _, f := range availableNonRootFolders {
+			childrenUIDs = append(childrenUIDs, f.UID)
+		}
+	}
+
 	dashFolders, err := s.dashboardFolderStore.GetFolders(ctx, cmd.OrgID, childrenUIDs)
 	if err != nil {
 		return nil, folder.ErrInternal.Errorf("failed to fetch subfolders from dashboard store: %w", err)
 	}
 
+	if cmd.UID == "" {
+		availableNonRootFoldersDedup := make([]*folder.Folder, 0)
+		for _, f := range availableNonRootFolders {
+			isSubfolder := slices.ContainsFunc(availableNonRootFolders, func(folder *folder.Folder) bool {
+				return f.ParentUID == folder.UID
+			})
+
+			if !isSubfolder {
+				parents, err := s.GetParents(ctx, folder.GetParentsQuery{UID: f.UID, OrgID: f.OrgID})
```

Comment:
If we could change folders implementation to materialized path, this operation would be trivial (every folder record will contain full list of parents).